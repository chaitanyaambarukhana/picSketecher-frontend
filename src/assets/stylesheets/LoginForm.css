.text-input {
  width: 80%;
  padding: 18px 45px;
  border: 1px solid #c2c2c2;
  border-radius: 20px;
  margin-top: 2%;
  font-size: medium;
}
.form-fields {
  padding: 5px 20px;
}

.form-field {
  margin-top: 5%;
}

.label {
  color: black;
  font-weight: 600;
}

.button-1 {
  background-color: #365a0c;
  border: none;
  border-radius: 20px;
  padding: 18px 45px;
  color: white;
  font-size: large;
  cursor: pointer;
  font-weight: 500;
  width: 100%;
  margin-top: 5%;
}

.forgot-password {
  display: flex;
  cursor: pointer;
  margin-left: 65%;
  margin-top: 2%;
}

.footer {
  text-align: center;
}

.sign-up-now {
  color: #365a0c;
  border: none;
  background: white;
  cursor: pointer;
  font-size: large;
  font-weight: bold;
}

/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 600px) {
  .form-field {
    margin-top: 3%;
  }

  .label {
    font-size: 1rem;
  }

  .text-input {
    padding: 12px 38px;
  }

  .button-1 {
    padding: 12px 38px;
    font-size: 1rem;
    width: 105%;
  }
  .forgot-password {
    margin-left: 61%;
    font-size: 0.8rem;
  }
  .footer {
    font-size: 1rem;
  }

  .sign-up-now {
    font-size: 1rem;
  }
}

/* Small devices (portrait tablets and large phones, 600px and up) */
@media only screen and (max-width: 767px) and (min-width: 600px) {
  .form-field {
    margin-top: 3%;
  }

  .label {
    font-size: 1rem;
  }

  .text-input {
    padding: 12px 38px;
  }

  .button-1 {
    padding: 12px 38px;
    font-size: 1rem;
    width: 100%;
  }
  .forgot-password {
    margin-left: 62%;
    font-size: 0.9rem;
  }
  .footer {
    font-size: 1rem;
  }

  .sign-up-now {
    font-size: 1rem;
  }
}

/* Medium devices (landscape tablets, 768px and up) */
@media only screen and(max-width: 991px) and (min-width: 768px) {
}

/* Large devices (laptops/desktops, 992px and up) */
@media only screen and (max-width: 1199px) and (min-width: 992px) {
}

/* Extra large devices (large laptops and desktops, 1200px and up) */
@media only screen and (max-width: 1399px) and (min-width: 1200px) {
  .forgot-password {
    margin-left: 62%;
    font-size: 1rem;
  }
}
