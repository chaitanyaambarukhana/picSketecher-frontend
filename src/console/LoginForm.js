import React, { useState } from "react";
import { Link } from "react-router-dom";
import "../assets/stylesheets/LoginForm.css";
import { titleCase } from "title-case";
const LoginForm = ({ handleTabChange, handleModalLoginClose }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [message, setMessage] = useState("");
  const [error, setError] = useState("");

  // const handleChange = (event) => {
  //   setEmail(event.target.value);
  //   setPassword(event.target.value);
  // };

  const handleLogin = (event) => {
    event.preventDefault();
    console.log("Login Button clicked");
    const request = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        email: email,
        password: password,
      }),
    };
    fetch("http://localhost:8000/login/", request)
      .then((res) => res.json())
      .then((data) => {
        if (data.success == true) {
          console.log("Success");
          setError("");
          setMessage(data.message);
          setEmail("");
          setPassword("");
          localStorage.setItem("jwt-token", data.jwt);
          handleModalLoginClose(false);
        } else {
          console.log("Error");
          setMessage("");
          setError(data.message);
        }
      });
  };

  const handleSignUp = () => {
    handleTabChange(true);
  };

  return (
    <div className="form-fields">
      {/* {message && <div className="success">{titleCase(message)}</div>} */}
      {error && <div className="error">{titleCase(error)}</div>}
      <div className="form-field">
        <label htmlFor="email" id="email" className="label">
          Email
        </label>
        <input
          type="email"
          className="text-input"
          id="email"
          value={email}
          onChange={(event) => {
            setEmail(event.target.value);
          }}
        />
      </div>
      <div className="form-field">
        <label htmlFor="password" id="password" className="label">
          Password
        </label>
        <input
          type="password"
          className="text-input"
          id="password"
          value={password}
          onChange={(event) => {
            setPassword(event.target.value);
          }}
        />
      </div>
      <button className="button-1" onClick={handleLogin}>
        LOGIN
      </button>
      <div style={{ display: "flex", justifyContent: "flex-end" }}>
        <a to="#" className="forgot-password">
          Forgot password ?
        </a>
      </div>
      <p className="footer">
        Not a member?{" "}
        <button onClick={handleSignUp} className="sign-up-now">
          Sign up now!
        </button>
      </p>
    </div>
  );
};
export default LoginForm;
