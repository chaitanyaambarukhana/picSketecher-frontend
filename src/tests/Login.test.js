import "jsdom-global/register";
import { shallow, mount } from "enzyme";
import "../../enzymeConfig.js";
import React from "react";
import renderer from "react-test-renderer";
import LoginForm from "../console/LoginForm";

describe("<Login/>", () => {
  const elem = renderer.create(<LoginForm />).toJSON();
  it("should match the snapshot", () => {
    expect(elem).toMatchSnapshot();
  });

  it("should render a email input tag", () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper.find('input[name="email"]').exists()).toBe(true);
  });

  it("should render a password input tag", () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper.find('input[name="password"]').exists()).toBe(true);
  });

  it("should render a submit button", () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper.find('button[name="button"]').exists()).toBe(true);
  });

  it("the default value for both fields should be empty", () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper.find('input[name="email"]').prop("value")).toBe("");
    expect(wrapper.find('input[name="password"]').prop("value")).toBe("");
  });

  it("on change of value in the field, the state of that field in the component should be updated", () => {
    const wrapper = shallow(<LoginForm />);

    wrapper.find('input[name="email"]').simulate("change", {
      target: {
        value: "email@gmail.com",
      },
    });
    expect(wrapper.find('input[name="email"]').prop("value")).toBe(
      "email@gmail.com"
    );
    wrapper.find('input[name="password"]').simulate("change", {
      target: {
        value: "Test@1234",
      },
    });
    expect(wrapper.find('input[name="password"]').prop("value")).toBe(
      "Test@1234"
    );
  });

  it("on submitting, a submit handler function should be triggered on click event", () => {
    const fn = jest.fn();
    const wrapper = mount(<LoginForm onSubmit={fn} />);
    wrapper.find("form").simulate("click");
    expect(fn.mock.calls.length).toEqual(0);
  });
});
