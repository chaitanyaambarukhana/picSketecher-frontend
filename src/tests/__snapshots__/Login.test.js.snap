// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should match the snapshot 1`] = `
<form
  className="form-fields"
  onSubmit={[Function]}
>
  
  <div
    className="form-field"
  >
    <label
      className="label"
      htmlFor="email"
      id="email"
    >
      Email
    </label>
    <input
      className="text-input"
      id="email"
      name="email"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="label"
      htmlFor="password"
      id="password"
    >
      Password
    </label>
    <input
      className="text-input"
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    className="button-1"
    id="button"
    name="button"
  >
    LOGIN
  </button>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-end",
      }
    }
  >
    <a
      className="forgot-password"
      to="#"
    >
      Forgot password ?
    </a>
  </div>
  <p
    className="footer"
  >
    Not a member?
     
    <button
      className="sign-up-now"
      onClick={[Function]}
    >
      Sign up now!
    </button>
  </p>
</form>
`;
